src/System.Threading.Tasks/tests/Task/TaskCancelWaitTests.cs

```diff
@@ -2748,6 +2748,38 @@ public static void TaskCancelWait141()
             test.RealRun();
         }
 
+        [Fact]
+        [OuterLoop]
+        public static void TaskCancelWait142()
+        {
+            // Verify that Task.Wait can return before all continuations scheduled
+            // with ExecuteSynchronously complete
+
+            Task task1 = new Task(() =>
+            {
+                Task.Delay(1).Wait();
+            });
+
+            CancellationTokenSource cts = new CancellationTokenSource();
+            Task task2 = task1.ContinueWith((_task) =>
+            {
+                while (!cts.IsCancellationRequested)
+                {
+                    Task.Delay(1).Wait();
+                }
```

Comment:
This could just be:
```C#
cts.WaitHandle.WaitOne();
```
or rather than using a CTS, a ManualResetEventSlim could be used directly.